sharding/xorValidatorShuffler.go

```diff
@@ -102,20 +102,28 @@ func (rxs *randXORShuffler) UpdateNodeLists(args ArgsUpdateNodes) (map[uint32][]
 }
 
 func removeFromWaiting(waitingNodes map[uint32][]Validator, leavingNodes []Validator) (map[uint32][]Validator, []Validator) {
```

Comment:
is this only computing or computing and removing? Doesn't seem right that a function called removeFromWaiting does 2 computing and no actual removing.

Reply:
renamed internal method removeNodes